src/main/java/org/jooq/lambda/Seq.java

```diff
@@ -203,11 +203,27 @@
     }
 
     /**
+     * Cross join stream with itself into one.
+     * <p>
+     * <code><pre>
+     * // (tuple(1, 1), tuple(1, 2), tuple(2, 1), tuple(2, 2))
+     * Seq.of(1, 2).crossSelfJoin()
+     * </pre></code>
+     */
+    default Seq<Tuple2<T, T>> crossSelfJoin() {
+
+        // This algorithm isn't lazy and has substantial complexity for large argument streams!
+        List<? extends T> list = toList();
+
+        return Seq.crossJoin(seq(list), seq(list));
+    }
+
+    /**
      * Inner join 2 streams into one.
      * <p>
      * <code><pre>
      * // (tuple(1, 1), tuple(2, 2))
-     * Seq.of(1, 2, 3).innerJoin(Seq.of(1, 2), t -> Objects.equals(t.v1, t.v2))
+     * Seq.of(1, 2, 3).innerJoin(Seq.of(1, 2), (t, u) -> Objects.equals(t, u))
```

Comment:
Thanks for these. Would you mind fixing these in a separate pull request, as they're not strictly related to this particular change...